services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -517,28 +517,45 @@ async def delete_file(
         # Only use this in those circumstances where a collaborator requires to delete a file (the current
         # permissions model will not allow him to do so, even though this is a legitimate action)
         # SEE https://github.com/ITISFoundation/osparc-simcore/issues/5159
+        def _get_subpath(path: str, levels: int):
+            components = path.strip("/").split("/")
+            subpath = "/".join(components[:levels])
+            return "/" + subpath if subpath else "/"
+
```

Comment:
This should be in the simcore_s3_utils.py with some appropriate test with parameters such as:
```python
@pytest.mark.parametrize("file_id, expected_file_id_prefix", [(f"b21a3b80-d578-4b33-a224-e24ee2e4966a/42b9cc07-60f5-4d29-a063-176d1467901c/my/amazing/sub/folder/with/a/file.bin", "b21a3b80-d578-4b33-a224-e24ee2e4966a/42b9cc07-60f5-4d29-a063-176d1467901c/my"), (f"api/42b9cc07-60f5-4d29-a063-176d1467901c/my/amazing/sub/folder/with/a/file.bin", "api/42b9cc07-60f5-4d29-a063-176d1467901c/my")])
def test_compute_partial_prefix(file_id, expected_file_id_prefix):
    assert compute_partial_prefix(file_id) == expected_file_id_prefix
```